internal/git/provider/provider.go

```diff
@@ -48,27 +51,17 @@ func FromURL(repoURL string, options ...ProviderOption) (Provider, Repository, e
 	targetProvider := ""
 	repo := Repository{}
 
-	parsedURL, err := url.Parse(repoURL)
+	gitURL, err := giturl.NewGitURL(repoURL)
 	if err != nil {
-		return nil, repo, fmt.Errorf("unable to parse url: %w", err)
+		return nil, repo, fmt.Errorf("failed parsing repository url: %w", err)
 	}
 
-	// That's a pretty naiv check for now.
-	// Use proper parsing later with a well tested library.
-	if parsedURL.Hostname() == "github.com" {
-		targetProvider = GitHubProviderName
-
-		parts := strings.Split(parsedURL.Path, "/")
-		if len(parts) != 3 {
-			return nil, repo, fmt.Errorf("invalid github repository url: %s", repoURL)
-		}
-
-		repo.Org = parts[1]
-		repo.Name = strings.TrimSuffix(parts[2], ".git")
-	}
+	targetProvider = gitURL.GetProvider()
+	repo.Org = gitURL.GetOwnerName()
+	repo.Name = gitURL.GetRepoName()
 
-	if targetProvider == "" {
-		return nil, repo, fmt.Errorf("could not parse provider from url: %s", repoURL)
+	if targetProvider == AzureProviderName {
```

Comment:
The range of `GetProvider` is the consts in https://github.com/kubescape/go-git-url/blob/master/apis/provider.go, not the consts defined here. This comparison should really use the const from the library, otherwise it relies on the coincidence of the strings being the same.

Reply:
That works, good thinking.